pkg/k8s/client.go

```diff
@@ -321,6 +262,15 @@ func (c *Client) ListIngressBackendPolicies() []*policyv1alpha1.IngressBackend {
 	return backends
 }
 
+// GetExtensionService returns the extension service for the given service ref
+func (c *Client) GetExtensionService(svc policyv1alpha1.ExtensionServiceRef) *configv1alpha2.ExtensionService {
```

Comment:
I'd consider changing this to accept a name, namespace string

Reply:
I'd keep this as is for convenience of directly passing the ExtensionServiceRef from the Telemetry resource.